calico_cni_test.go

```diff
@@ -254,6 +260,49 @@ var _ = Describe("CalicoCni", func() {
 		})
 	})
 
+	Context("With IP forwarding enabled", func() {
+		netconf := fmt.Sprintf(`
+			{
+			  "cniVersion": "%s",
+			  "name": "net1",
+			  "type": "calico",
+			  "etcd_endpoints": "http://%s:2379",
+			  "log_level": "debug",
+			  "nodename_file_optional": true,
+			  "datastore_type": "%s",
+			  "container_settings": {
+			    "allow_ip_forwarding": true
+			  },
+			  "ipam": {
+			    "type": "host-local",
+			    "subnet": "10.0.0.0/8"
+			  }
+			}`, cniVersion, os.Getenv("ETCD_IP"), os.Getenv("DATASTORE_TYPE"))
+
+		It("should enable IPv4 forwarding", func() {
+			containerID := fmt.Sprintf("con%d", rand.Uint32())
+			_, session, _, _, _, contNs, err := testutils.CreateContainerWithId(netconf, "", testutils.TEST_DEFAULT_NS, "", containerID)
+
+			By("successfully networking the container", func() {
+				Expect(err).ShouldNot(HaveOccurred())
+				Eventually(session).Should(gexec.Exit(0))
+			})
+
+			By("asserting IPv4 forwarding is enabled", func() {
+				targetNs, _ := ns.GetNS(contNs.Path())
+				err = targetNs.Do(func(_ ns.NetNS) error {
+					return testutils.CheckSysctlValue("/proc/sys/net/ipv4/ip_forward", "1")
+				})
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			By("tearing down the container", func() {
+				_, err = testutils.DeleteContainerWithId(netconf, contNs.Path(), "", testutils.TEST_DEFAULT_NS, containerID)
```

Comment:
should this be in a `defer`?

Reply:
Probably not a defer, but perhaps it belongs in an `AfterEach` clause.